### PR TITLE
chore: bump `@webassemblyjs`

### PR DESCRIPTION
**Pull request by alexander-akait**
Bump `@webassemblyjs` because the bug was fixed + small more update

### <samp>🤖 Generated by Copilot at f9bb49c</samp>

This pull request updates `package.json` to enhance WebAssembly and schema validation functionality and remove unused code.

### <samp>🤖 Generated by Copilot at f9bb49c</samp>

* Update `@webassemblyjs` dependencies to fix bugs and improve performance ([link](https://github.com/webpack/webpack/pull/17003/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L10-R12),[link](https://github.com/webpack/webpack/pull/17003/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L106-R105))
* Update `schema-utils` dependency to fix validation issues and support new options ([link](https://github.com/webpack/webpack/pull/17003/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L27-R27))
* Remove unnecessary `@types/es-module-lexer` devDependency ([link](https://github.com/webpack/webpack/pull/17003/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L41))
* Update `jest-junit` devDependency to support Jest 27 and improve output format ([link](https://github.com/webpack/webpack/pull/17003/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L74-R73))

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,7 +24,7 @@
     "loader-runner": "^4.2.0",
     "mime-types": "^2.1.27",
     "neo-async": "^2.6.2",
-    "schema-utils": "^3.1.0",
+    "schema-utils": "^3.1.2",
```
Comment (alexander-akait) [inline package.json:27]: I implemented lazy Ajv loazing, so initial startup is faster

File: package.json [hunk 2]
```diff
@@ -38,7 +38,6 @@
   "devDependencies": {
     "@babel/core": "^7.21.4",
     "@babel/preset-react": "^7.18.6",
-    "@types/es-module-lexer": "^0.4.1",
```
Comment (alexander-akait) [inline package.json:41]: No need, because it is `stub`, `es-module-lexer` has types own types right now

File: package.json [hunk 3]
```diff
@@ -71,7 +70,7 @@
     "jest-cli": "^29.5.0",
     "jest-diff": "^29.5.0",
     "jest-environment-node": "^29.5.0",
-    "jest-junit": "^15.0.0",
+    "jest-junit": "^16.0.0",
```
Comment (alexander-akait) [inline package.json:73]: Just an update

File: package.json [hunk 4]
```diff
@@ -103,7 +102,7 @@
     "ts-loader": "^9.4.2",
     "typescript": "^4.8.4",
     "url-loader": "^4.1.0",
-    "wast-loader": "^1.11.0",
+    "wast-loader": "^1.11.5",
```
Comment (alexander-akait) [inline package.json:105]: Just an update too

**Review by TheLarkInn — APPROVED** [review]

**Comment by TheLarkInn** [comment]
Just a note: also should close #16998

**Pull request merged**